File: 3455-minimum-length-of-string-after-operations/minimum-length-of-string-after-operations.java
class Solution {
    public int minimumLength(String s) {

        Map<Character, Integer> map = new HashMap<>();
        int left = 0;
        int right = 0;
        int minLength = Integer.MAX_VALUE;

        while (right < s.length()) {
            map.put(s.charAt(right), map.getOrDefault(s.charAt(right), 0) + 1);

            if (map.get(s.charAt(right)) == 3) {
                int i = right;
                char ch = s.charAt(i);

                while (map.get(ch) > 1 && i >= 0) {
                    if (s.charAt(i) == ch) {
                        map.put(s.charAt(i), map.get(s.charAt(i)) - 1);
                    }
                    i--;
                }
            }
            right++;
        }

        StringBuilder ans = new StringBuilder();
        for (Map.Entry<Character, Integer> entry : map.entrySet()) {
            for (int i = 0; i < entry.getValue(); i++) {
                ans.append(entry.getKey());
            }
        }
        return ans.length();
        
    }
}

File: 3455-minimum-length-of-string-after-operations/minimum-length-of-string-after-operations.cpp
class Solution {
public:
    int minimumLength(string s) {
        unordered_map<char,int>mp;
        int k=2;
        int left=0;
        int right=0;
        int mini=INT_MAX;
        while(right<s.size()){
            mp[s[right]]++;
            if(mp[s[right]]==3){
                int i=right;
                char ch=s[i];
            while(mp[ch]>1 && i>=0){
               if(s[i]==ch){
                  mp[s[i]]--;

               }
              
                
                i--;
            }
          
        }
            
                right++;
            }
            string ans="";
            for(auto it:mp){
                for(int i=0;i<it.second;i++){
                    ans+=it.first;
                }
            }
            return ans.size();
            
        
        if(mini==INT_MAX){
            return s.size();
        }
        return mini;
        
    }
};